tests/multiple_database/models.py

```diff
@@ -80,3 +80,11 @@ class UserProfile(models.Model):
 
     class Meta:
         ordering = ('flavor',)
+
+
+class ForeignRelation(models.Model):
```

Comment:
I don't think a new model is needed. There are existing models that have ForeignKey/ManyToManyField/OneToOneField that could be used. Also, there are a number of existing tests (e.g. `test_(foreign/o2o/m2m)_key_cross_database_protection` that may duplicate some of the added tests). In particular, any tests that pass prior to this fix should be added separately.